### ISSUE TITLE
Clean up the staging environment from zombie collections

### ISSUE THREAD

**Issue by ebezzi** [chanzuckerberg/single-cell-data-portal#1375]
Functional tests that are run against the staging environment (API and the new revision one) cannot do a proper cleanup since they publish collections and deletion is only available for private collections. Our options are:

1. Change the tests so that they have direct access to the database (not recommended - it adds dependencies)
2. Create a script to manually clean up the staging database. This is easy since all the collections are from the same owner (a test user). The script can be run manually, scheduled or run at each staging deployment.
3. Create an authorized endpoint that can delete public collections.
4. [Timmy] For the E2E test account, can we keep at least 10 public collections for FE E2E to test revisions please? Thank you! I.E., the test account needs 10 public collections in My Collections page. Thank you!

**Comment by atolopko-czi** [comment]
Would it be possible to run functional tests on a new/clean rdev env instead of staging? This would help ensure we have a "controlled" backend state for the tests. 

As FE E2E tests rely upon staging too, switching to rdev would depend upon [#435](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-infra/435.)

**Comment by metakuni** [comment]
> 3. Create an authorized endpoint that can delete public collections.

Looks like we'll need that delete collection endpoint soon anyway. See:
- https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/132

**Comment by atolopko-czi** [comment]
@brianraymor noticed that this story is possibly addressed by the recent env mirroring capability. This will be the case if we add `--delete` flag to s3 sync cmd.  We held off on doing that initially, but have this [TODO](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/scripts/mirror_s3_data.sh#L16-L19) in the wings. I think we can either commander this story to add that option to the mirroring, or close this story and make a new one for that.  The real work of adding the `--delete` flag is just making sure we're not deleting anything in dev or staging that engineers might care about.  I noticed there was "extra" data in some of the affected buckets, so took the safe route until we could confirm the deletions were ok. I suspect it's fine to delete, though.

**Comment by atolopko-czi** [comment]
@ebezzi bump: Why do we need to delete collections at all in staging? Mirroring will eventually clean up the staging env.
I'm also not clear on 4: `keep 10 public collections for FE E2E to test revisions`. How are the FE E2E tests setting these up?

**Comment by atolopko-czi** [comment]
Closing, since env mirroring should address this cleanup. If there are other concerns I'm missing, please open a new issue.
